### DISCUSSION TITLE
Add `getContractEvent`

### DISCUSSION THREAD

**Discussion by horsefacts** [wevm/viem#1233] [Idea / Feature Request] [answered]
Hi Viem! I have a small feature suggestion to make it easier to handle historical events.   

`watchContractEvent` has the nice property of falling back to polling `eth_getLogs` if an RPC provider doesn't support stateful filters. But there's no equivalent for historical events that first tries `getFilterLogs` and then falls back to `getLogs`. 

This isn't a huge deal to manage by hand, but it would be nice to have a historical query function with the same fallback behavior built in to the library.

(More context on our use case: we sync [historical events](https://github.com/farcasterxyz/hub-monorepo/blob/13b4144a59ac601ac3f8bc6eb67060299395eb8d/apps/hubble/src/eth/l2EventsProvider.ts#L654) in Farcaster hubs, supporting user provided RPC providers and have found that a number of them don't support filters on Optimism).

**Comment by jxom** [comment]
#1235 

![](https://media4.giphy.com/media/tpTOw6sljB2U/giphy.gif?cid=ecf05e47jci1yo98ohci1fiorwcnsc3e03binpq217lf344f&ep=v1_gifs_search&rid=giphy.gif&ct=g)

**Reply by jxom** [reply, marked as answer]
https://viem.sh/docs/contract/getContractEvents.html

**Reply by horsefacts** [reply]
![aladdin-disney](https://github.com/wagmi-dev/viem/assets/109845214/c73b7c0e-923b-4820-b7a1-e0eec2f320dd)
